src/run_part.F90

```diff
@@ -442,6 +442,248 @@ subroutine pmc_mpi_unpack_run_part_opt(buffer, position, val)
 
   end subroutine pmc_mpi_unpack_run_part_opt
 
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Read the specification for a run_part simulation from a spec file.
+  subroutine spec_file_read_run_part(file, run_part_opt, aero_data, &
+       aero_state_init, gas_data, gas_state_init, env_state_init, &
+       aero_dist_init, scenario, &
+#ifdef PMC_USE_CAMP
+       camp_core, photolysis, aero_state, &
+#endif
+       n_part, rand_init, do_init_equilibrate, do_restart)
+
+    !> Spec file.
+    type(spec_file_t), intent(inout) :: file
+    !> Monte Carlo options.
+    type(run_part_opt_t), intent(inout) :: run_part_opt
+    !> Aerosol data.
+    type(aero_data_t), intent(inout) :: aero_data
+    !> Initial aerosol state.
+    type(aero_state_t), intent(inout) :: aero_state_init
+    !> Gas data.
+    type(gas_data_t), intent(inout) :: gas_data
+    !> Initial gas state.
+    type(gas_state_t), intent(inout) :: gas_state_init
+    !> Initial environmental state.
+    type(env_state_t), intent(inout) :: env_state_init
+    !> Initial aerosol distribution.
+    type(aero_dist_t), intent(inout) :: aero_dist_init
+    !> Scenario data.
+    type(scenario_t), intent(inout) :: scenario
+#ifdef PMC_USE_CAMP
+    !> CAMP core.
+    type(camp_core_t), pointer :: camp_core
+    !> Photolysis calculator.
+    type(photolysis_t), pointer :: photolysis
+    !> Aerosol state.
+    type(aero_state_t), intent(inout) :: aero_state
+#endif
+    !> Ideal number of computational particles.
+    real(kind=dp), intent(inout) :: n_part
+    !> Random number generator seed.
+    integer, intent(out) :: rand_init
+    !> Whether to equilibrate.
+    logical, intent(out) :: do_init_equilibrate
+    !> Whether simulation is a restart.
+    logical, intent(out) :: do_restart
+
+    integer :: i_repeat, i_group
+    logical :: read_aero_weight_classes
+    character(len=PMC_MAX_FILENAME_LEN) :: restart_filename
+    integer :: dummy_index, dummy_i_repeat
+    real(kind=dp) :: dummy_time, dummy_del_t
+    character(len=PMC_MAX_FILENAME_LEN) :: sub_filename
+    type(spec_file_t) :: sub_file
+    character(len=PMC_MAX_FILENAME_LEN) :: camp_config_filename
+
+    if (pmc_mpi_rank() == 0) then
```

Comment:
I think it's a bit confusing the way the new `spec_file_read_run_part()` function only reads the spec file on the rank-0 MPI process, but then it relies on the caller code to broadcast the spec file data to all other processes. I think we should either:
1. Also move the "send to other processes" code into `spec_file_read_run_part()`.
2. Or make `spec_file_read_run_part()` only do the spec file reading and add another function that broadcasts the spec file to other processes, so in `partmc_part()` we would do something like:
    ```
    if (pmc_mpi_rank() == 0) then
        call spec_file_read_run_part()
    end if
    call spec_file_broadcast_run_part()
    ```

I think that option (1) would be the simplest rearrangement of the existing code, but it does mean that we don't have much flexibility with the MPI handling. Option (2) is more complex but might give us more flexibility in the future. I could go either way on this and I think either of these options is fine.

Reply:
I think the flexibility is why I left ultimately it where it was... so I think I'll implement option (2).

Reply:
Looks good. With the new structure, I think that `spec_file_read_run_part()` should no longer have the `if (pmc_mpi_rank() == 0)` in it? Because we only call this function if we are rank zero.